ofrak_core/ofrak/model/viewable_tag_model.py

```diff
@@ -16,19 +16,37 @@
 
 _VIEW_ATTRIBUTES_TYPE = "__view_attributes_type__"
 _COMPOSED_ATTRIBUTES_TYPE = "__composed_attributes_types__"
-_VIEW_ATTRIBUTE_TYPE_NAME_SUFFIX = "AutoAttributes"
 
 
 RA = TypeVar("RA", bound=ResourceAttributes)
 RVI = TypeVar("RVI", bound="ResourceViewInterface")
 
 
 class AttributesType(ResourceAttributes, Generic[RVI]):
+    """
+    A Generic type for `ViewableResourceTag` to get the auto-generated `ResourceAttributes`
+    class associated with that view type. The returned class is a `dataclass` which encapsulates
+    the fields defined in one specific `ViewableResourceTag`.
+    For example if `B` inherits from `A` and `A` defines one or more new fields,
+    `AttributesType[B]` has only fields defined in `B`, and none of the fields defined in `A`.
+    """
+
     def __init__(self, *args: Any, **kwargs: Any):
         raise NotImplementedError("AttributesType cannot be instantiated")
 
     def __class_getitem__(cls, item: Type[RVI]) -> Type[ResourceAttributes]:
-        return item.attributes_type
+        return getattr(item, _VIEW_ATTRIBUTES_TYPE)
+
+    def __getattr__(self, item):  # pragma: no cover
+        """
+        Type stub that 'tricks' MyPy into not trying to typecheck attribute accesses of an
+        `AttributesType` instance. This stub solves the following case:
+        x: AttributesType[X] = func()
+        x.any_field  <--- Always is a type error because MyPy thinks AttributesType has no fields!
```

Comment:
````suggestion

        ```
        x: AttributesType[X] = func()
        x.any_field  <--- Always is a type error because MyPy thinks AttributesType has no fields!
        ```
````